### PR TITLE
Fuzzing: add more fuzzers

### PR DESCRIPTION
**Pull request by AdamKorcz**
Signed-off-by: AdamKorcz <adam@adalogics.com>

#### Summary

Adds three more fuzzers.

### CODE REVIEW

File: pkg/signer/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package signer
+
+import (
+	"os"
+	"testing"
+)
+
+func FuzzNewFile(f *testing.F) {
+	f.Fuzz(func(t *testing.T, keyData []byte, keyPass string) {
+		f, err := os.Create("keyfile")
```
Comment (bobcallaway) [inline pkg/signer/fuzz_test.go]: should this be `os.CreateTemp`?

File: pkg/signer/fuzz_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			t.Skip()
+		}
+		_, err = f.Write(keyData)
+		if err != nil {
+			t.Skip()
+		}
+		defer func() {
+			f.Close()
+			os.Remove("keyfile")
+		}()
```
Comment (bobcallaway) [inline pkg/signer/fuzz_test.go]: should this be moved up above the call to `f.Write()`?

File: pkg/signer/fuzz_test.go (hunk 1 above)
Comment (kommendorkapten) [reply pkg/signer/fuzz_test.go]: Or `t.TempDir()`? As that directory is automatically cleaned up.

Comment (kommendorkapten) [reply pkg/signer/fuzz_test.go]: I.e create the file in the temporary directory :)

Comment (AdamKorcz) [reply pkg/signer/fuzz_test.go]: > should this be `os.CreateTemp`?

Done and rebased.

File: pkg/signer/fuzz_test.go (hunk 2 above)
Comment (AdamKorcz) [reply pkg/signer/fuzz_test.go]: Done and rebased.

File: pkg/signer/fuzz_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+package signer
+
+import (
+	"os"
+	"testing"
+)
+
+func FuzzNewFile(f *testing.F) {
+	f.Fuzz(func(t *testing.T, keyData []byte, keyPass string) {
+		keyPath, err := os.CreateTemp("", "keyfile")
```
Comment (bobcallaway) [inline pkg/signer/fuzz_test.go]:
```suggestion
		keyPath, err := os.CreateTemp(t.TempDir(), "keyfile")
```

Comment (AdamKorcz) [reply pkg/signer/fuzz_test.go]: Done and rebased.

**Review by bobcallaway — APPROVED** [review]

File: pkg/signer/fuzz_test.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+func FuzzNewFile(f *testing.F) {
+	f.Fuzz(func(t *testing.T, keyData []byte, keyPass string) {
+		keyPath, err := os.CreateTemp(t.TempDir(), "keyfile")
+		if err != nil {
+			t.Skip()
+		}
+		defer func() {
+			keyPath.Close()
+			os.Remove(keyPath.Name())
```
Comment (kommendorkapten) [inline pkg/signer/fuzz_test.go:32]: You don't need to call `os.Remove` as the directory (and it's contents) created by `t.TempDir()` is removed automatically when the test is finished.

File: pkg/signer/fuzz_test.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+package signer
+
+import (
+	"os"
+	"testing"
+)
+
+func FuzzNewFile(f *testing.F) {
+	f.Fuzz(func(t *testing.T, keyData []byte, keyPass string) {
+		keyPath, err := os.CreateTemp(t.TempDir(), "keyfile")
```
Comment (kommendorkapten) [inline pkg/signer/fuzz_test.go:26]: Maybe rename `keyPath` to `keyFile` as `CreateTemp` returns a `*File`, not a path.

**Pull request merged**